### DISCUSSION TITLE
SonOff Basic R2 ESP8266EX can't connect to WiFi after followed upgrade-path

### DISCUSSION THREAD

**Discussion by exetico** [arendst/Tasmota#22417] [General]
Hi,

I dont have the proper time to find JTAG information on the R2 ESP8266EX  right now, so I'll start with a discussion.

My SonOff have not been in use for a good period, but a few weeks ago I started using some of them, again.

Therefore I opted to upgrade all of them to the latest Tasmota version (14.3.0). 

All SonOff Basic R2 ESP8285N08 did that without problems, following the recommended upgrade-path([v6.7.1](http://ota.tasmota.com/tasmota/release-6.7.1/sonoff.bin)  🔀  [v7.2.0](http://ota.tasmota.com/tasmota/release-7.2.0/tasmota-lite.bin)  🔀  [v8.5.1](http://ota.tasmota.com/tasmota/release-8.5.1/tasmota-lite.bin)  🔀  [v9.1](http://ota.tasmota.com/tasmota/release-9.1.0/tasmota-lite.bin.gz)  🔀  [Current release](http://ota.tasmota.com/tasmota/release/)), however, three of my SonOff Basic R2 ESP8266EX didn't like it at all.

After the upgrade to 14.3, they didn't reconnect to the WiFi.

After a network-reset like (done like described in the docs for 14.3) the captive portal worked, **if** I typed in a static IP on my phone. My IoT WiFi was visible in the network list (yes, 2.4GHz are isolated on it's on SSID). But, no matter what I did, they could not connect to it.

I've not extracted any logging from the devices itself, but I upgraded them, one by one. I tried different things like using a lite image mid-way, and so on.

However, maybe other users should be alerted about the ESP8266EX situation on SonOff Basic R2?

Here's the screenshots after the upgrade to 14.3.0, captured in the hotspot-mode:

![image](https://github.com/user-attachments/assets/d4d5b316-4c4b-4960-846d-a8485f932897)
![image](https://github.com/user-attachments/assets/86e07a0f-5b26-4e0f-9405-c6c0ab3a3846)

**UPDATE**:

While making some clean-up's I actually noticed that **two** of the four I have on 14.3.0, are `ESP8266EX` based, too. I'm wondering if I have some ESP8266EX R2 clones, or if some of my devices originally was flashed in another way, and that make the difference between a succesful upgrade to 14.3.0, or not.

Below are one of the ESP8266EX R2's which did handle the upgrade to 14.3.0, and did manage to connect to my wifi without problems, other than entering the information in the captive portal:

<img width="420" alt="image" src="https://github.com/user-attachments/assets/06f8ae30-9fdc-4d2e-b33d-ffcda7080c8e">

```

**Comment by exetico** [comment]
Okay, it's wierd. On two of the SonOff Basic PCB's there's printed "R2", but they looks like the R1 design to me:

https://tasmota.github.io/docs/devices/Sonoff-Basic/#sonoff-basic-r1

I've updated my initial post.

Below are the image of three devices. There's nothing below the white QC sticker, other than a "Sonoff" text.

 Now I'm considering checking the two working ESP8266EX's, and see what kind of layout and information they have printed on them.

![image](https://github.com/user-attachments/assets/6b12f921-d267-463f-b20c-955d983b7c56)

<img width="870" alt="image" src="https://github.com/user-attachments/assets/b72a6a88-aa8a-4f4d-bc1c-b5a9de822adb">

_One of them are still sitting on a older firmware._

**Reply by sfromis** [reply]
Well, old Sonoff Basic are crappy devices, not least the R2. Known to suddenly fail, I've seen that twice. They are barely able to power themselves, but sometimes staying on older Tasmota versions (less features) seems to help with old-age frailty..... Not related to the MCU chip as such, more like the rest of the boards.

Hard to see, but *maybe* a couple of the lyts look a bit dubious.

**Reply by exetico** [reply]
It's sure not anything new, but I've not had issues with them before. With that said: Lots of great things have been introduced since SonOff released the Basic model.

What kind of Tasmota-supported device do you recommend today?

Also, I'm positive about the thought, that it's possibly just a bad device missing the proper amount of power, or whatever. I'm considering replacing the other three devices in the house.

The only other thing I need to mention, is you're skipping the fact that the docs state that R2 are expected to have the new trace-design, but two of three "R2" marked devices looks like the R1 device. So, possible two fake-devices, right?

**Reply by Wintespe** [reply]
These are not fakes, but original Sonoff R1 Basic with the hardware sub-release R2. They have nothing to do with the new Sonoff Basic R2, because they have a completely different design. I still have these quite old parts from 2017 in operation. However, I see no reason to change the software version (8.5.1), because the things do what they are supposed to: turn a relay on/off, nothing more. Since they are now 7 years old, I don't think it's a bad thing if a part like that for €3.50 dies in old age.

**Comment by crhass** [comment]
I have some of these sonoff basics with the R2 hardware. I have found they work ok with Tasmota v12.2.0. Last time I tried anything higher they would not connect to WiFi.
I also have some with DHT22 temp/humidity sensors with no issues.

**Reply by sfromis** [reply]
That's the pain with those *crap* devices. With borderline hardware, it is a gamble about when they can't keep up.

**Comment by exetico** [comment]
I'm not sure if I should agree or not, regarding the state of the SonOff R2's. They have served me well, and I didn't fell that it's a gamble. But, I'll close is, just to reflect that it's not really a surprise. With that said, I can't deny if I looked at the other esp.... home ... solution, and found that to handle them better, of if I swapped all of them, with new devices 😄 .

**Comment by pezoli** [comment]
Bit late to join the party ... but i had similar issue upgrade my Sonoff Basic R1 (R2 on the board) from v8.3.1. I was using full firmware (tasmota.bin). Wanted to install the latest release (v15.2.0), followed documented upgrade path, tasmota-minimal.bin and tasmota-lite.bin was working fine, but after upgrading to tasmota.bin, wifi couldn't connect. I have tried with different sequence, erased before flashing, used esptools and tasmotizer 1.2, always same result, full firmware cant connect to the wifi. SSID and credentials was same for tasmota-lite.bin and tasmota.bin, still the latter did not connect. I used putty serial monitor to see what is happening, it was kept trying to connect but failed. Something like this:
`00:01:41.768 WIF: Connecting to AP1 MYSSID in mode 11n as tasmota-17E871-2661...`
`00:01:52.771 WIF: Connect failed with AP timeout`
Finally i decided to stay on lite firmware, this works for me as i am not using sensors.

**Reply by r3wald** [reply]
> Finally i decided to stay on lite firmware, this works for me as i am not using sensors.

Thank you very much for this tip. I have a couple of old Sonoffs which were no longer responding after I updated them to Tasmota 14+. The lite-variant works though.

**Reply by sfromis** [reply]
Better have a plan ready to repair/replace. Sounds like those devices are degraded, but not quite failed yet.

**Reply by exetico** [reply]
Sure, personally I've swapped them out by now, after listening to the feedback.
